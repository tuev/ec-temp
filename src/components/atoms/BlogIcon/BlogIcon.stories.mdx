import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './BlogIcon.stories.tsx'
import BlogIcon from './BlogIcon.tsx'

<Meta title="BlogIcon" component={BlogIcon} />
