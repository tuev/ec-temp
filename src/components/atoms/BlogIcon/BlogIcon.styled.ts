import { COLOR } from 'theme/colors'
import styled from 'styled-components'

export const AppBlogIconWrapper: React.FC = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
`

export const AppBlogIconCircle: React.FC = styled.div`
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: ${COLOR.gray2};
  display: flex;
  align-items: center;
  justify-content: center;
`
