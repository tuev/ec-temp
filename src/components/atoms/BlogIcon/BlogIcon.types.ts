type BlogIconType = 'like' | 'comment' | 'bookmark' | 'options'

interface IBlogIcon extends React.HTMLProps<HTMLDivElement> {
  icon?: 'like' | 'comment' | 'bookmark' | 'options'
  count?: number
}

export declare type BaseBlogIconProps = IBlogIcon

export type GetIcon = (icon?: BlogIconType) => React.ReactNode
