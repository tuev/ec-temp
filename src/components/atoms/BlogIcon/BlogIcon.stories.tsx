import React from 'react'
import { action } from '@storybook/addon-actions'
import BlogIcon from './BlogIcon'

export default {
  title: 'BlogIcon',
  component: BlogIcon,
  includeStories: ['basic'],
}

export const basic = () => (
  <div style={{ width: 350, display: 'flex', justifyContent: 'space-between' }}>
    <BlogIcon onClick={action('BlogIcon Clicked!!!')} icon="like" count={20} />
    <BlogIcon
      onClick={action('BlogIcon Clicked!!!')}
      icon="comment"
      count={20}
    />
    <BlogIcon onClick={action('BlogIcon Clicked!!!')} icon="bookmark" />
    <BlogIcon onClick={action('BlogIcon Clicked!!!')} icon="options" />
  </div>
)
