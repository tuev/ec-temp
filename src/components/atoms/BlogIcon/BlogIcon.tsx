import React, { FC, useMemo } from 'react'
import { AppBlogIconWrapper, AppBlogIconCircle } from './BlogIcon.styled'
import { BaseBlogIconProps, GetIcon } from './BlogIcon.types'
import {
  FavoriteBorder,
  ChatBubbleOutline,
  BookmarkBorder,
  MoreHoriz,
} from '@material-ui/icons'

const getIcon: GetIcon = icon => {
  switch (icon) {
    case 'like':
      return <FavoriteBorder color="error" fontSize="small" />
    case 'comment':
      return <ChatBubbleOutline fontSize="small" />
    case 'bookmark':
      return <BookmarkBorder fontSize="small" />
    default:
      return <MoreHoriz fontSize="small" />
  }
}

const BlogIcon: FC<BaseBlogIconProps> = (props: BaseBlogIconProps) => {
  const { icon, count } = props
  const iconRender = useMemo(() => getIcon(icon), [icon])

  return (
    <AppBlogIconWrapper>
      <AppBlogIconCircle>{iconRender}</AppBlogIconCircle>
      {count ? <span>{count}</span> : <span />}
    </AppBlogIconWrapper>
  )
}

export default BlogIcon
