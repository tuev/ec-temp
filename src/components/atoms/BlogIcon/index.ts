import BlogIcon from './BlogIcon'

export { BlogIcon }

export default BlogIcon
