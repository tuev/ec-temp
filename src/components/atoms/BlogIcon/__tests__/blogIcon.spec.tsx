import React from 'react'
import { mount } from 'enzyme'
import BlogIcon from '..'

describe('BlogIcon', () => {
  it('like icon', () => {
    const likeIcon = mount(<BlogIcon icon="like" count={20} />)

    expect(likeIcon).toMatchSnapshot()
    const expectStyle: { [key: string]: string } = {
      'align-items': 'center',
      cursor: 'pointer',
      display: 'flex',
      'justify-content': 'space-around',
      'max-width': '80px',
    }

    for (const key in expectStyle) {
      if (expectStyle[key]) {
        expect(likeIcon).toHaveStyleRule(key, expectStyle[key])
      }
    }
  })

  it('comment icon', () => {
    const commentIcon = mount(<BlogIcon icon="comment" count={20} />)
    expect(commentIcon).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      display: 'flex',
      'align-items': 'center',
      'justify-content': 'space-around',
      'max-width': '80px',
      cursor: 'pointer',
    }

    for (const key in expectStyle) {
      if (expectStyle[key]) {
        expect(commentIcon).toHaveStyleRule(key, expectStyle[key])
      }
    }
  })

  it('bookmark icon', () => {
    const bookmarkIcon = mount(<BlogIcon icon="bookmark" />)
    expect(bookmarkIcon).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      display: 'flex',
      'align-items': 'center',
      'justify-content': 'space-around',
      'max-width': '80px',
      cursor: 'pointer',
    }

    for (const key in expectStyle) {
      if (expectStyle[key]) {
        expect(bookmarkIcon).toHaveStyleRule(key, expectStyle[key])
      }
    }
  })

  it('options icon', () => {
    const optionsIcon = mount(<BlogIcon icon="options" />)
    expect(optionsIcon).toMatchSnapshot()
    const expectStyle: { [key: string]: string } = {
      display: 'flex',
      'align-items': 'center',
      'justify-content': 'space-around',
      'max-width': '80px',
      cursor: 'pointer',
    }

    for (const key in expectStyle) {
      if (expectStyle[key]) {
        expect(optionsIcon).toHaveStyleRule(key, expectStyle[key])
      }
    }
  })
})
