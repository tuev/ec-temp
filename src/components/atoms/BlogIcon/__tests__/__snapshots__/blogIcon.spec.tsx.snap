// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogIcon bookmark icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
}

.c1 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #F2F2F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<BlogIcon
  icon="bookmark"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <ForwardRef
            fontSize="small"
          >
            <WithStyles(ForwardRef(SvgIcon))
              fontSize="small"
            >
              <ForwardRef(SvgIcon)
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                fontSize="small"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </div>
      </styled.div>
      <span />
    </div>
  </styled.div>
</BlogIcon>
`;

exports[`BlogIcon comment icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
}

.c1 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #F2F2F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<BlogIcon
  count={20}
  icon="comment"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <ForwardRef
            fontSize="small"
          >
            <WithStyles(ForwardRef(SvgIcon))
              fontSize="small"
            >
              <ForwardRef(SvgIcon)
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                fontSize="small"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </div>
      </styled.div>
      <span>
        20
      </span>
    </div>
  </styled.div>
</BlogIcon>
`;

exports[`BlogIcon like icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
}

.c1 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #F2F2F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
      />
    </svg>
  </div>
  <span>
    20
  </span>
</div>
`;

exports[`BlogIcon options icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
}

.c1 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #F2F2F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<BlogIcon
  icon="options"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <ForwardRef
            fontSize="small"
          >
            <WithStyles(ForwardRef(SvgIcon))
              fontSize="small"
            >
              <ForwardRef(SvgIcon)
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                fontSize="small"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </div>
      </styled.div>
      <span />
    </div>
  </styled.div>
</BlogIcon>
`;
